src/java.base/share/classes/java/util/Arrays.java

```diff
@@ -4032,9 +4122,24 @@ public static float[] copyOfRange(float[] original, int from, int to) {
      * @since 1.6
      */
     public static double[] copyOfRange(double[] original, int from, int to) {
+        // Tickle the JIT to fold special cases optimally
+        if (from != 0 || to != original.length)
+            return copyOfRangeGeneric(original, from, to);
+        else // from == 0 && to == original.length
+            return copyOf(original);
+    }
+
+    @ForceInline
+    private static double[] copyOf(double[] original) {
+        double[] copy = new double[original.length];
+        System.arraycopy(original, 0, copy, 0, original.length);
+        return copy;
+    }
+
+    @ForceInline
+    private static double[] copyOfRangeGeneric(double[] original, int from, int to) {
+        checkLength(from, to);
         int newLength = to - from;
```

Comment:
```suggestion
        int newLength = checkLength(from, to);
```

Reply:
> I'm also curious if returning the new length from `checkLength` would be worthwhile

I had this arrangement earlier but saw no win from it. These helper methods will be aggressively inlined and the JIT sorts it out optimally, so I prefer to keep the `check` method zoomed in on its purpose